src/Azure.Functions.Cli/Helpers/NetworkHelpers.cs

```diff
@@ -33,5 +37,33 @@ public static int GetAvailablePort()
                 listener.Stop();
             }
         }
+
+        public static int GetNextAvailablePort(int startPort)
+        {
+            var usedPorts = new List<int>();
+            var ipGlobalProperties = IPGlobalProperties.GetIPGlobalProperties();
+
+            usedPorts.AddRange(ipGlobalProperties
+                                .GetActiveTcpConnections()
+                                .Where(c => c.LocalEndPoint.Port >= startPort)
+                                .Select(c => c.LocalEndPoint.Port));
+            usedPorts.AddRange(ipGlobalProperties
+                                .GetActiveTcpListeners()
+                                .Where(l => l.Port >= startPort)
+                                .Select(l => l.Port));
+
+            usedPorts.Sort();
```

Comment:
Why sorting is needed for all used ports, thats an additional complexity of O(NlogN). You are iterating over all the values in usedPorts.

You have two options,
1. Used HashSet and iterate over all the values and lookup will be O(N), then you can do startPort++ and keep looking.
2. Used TreeSet or SortedSet to store and look for value less than startPort.

Reply:
Done! 👍